flake8_type_checking/checker.py

```diff
@@ -674,43 +677,33 @@ def register_function_annotations(self, node: Union[FunctionDef, AsyncFunctionDe
         for path in [node.args.args, node.args.kwonlyargs, node.args.posonlyargs]:
             for argument in path:
                 if hasattr(argument, 'annotation') and argument.annotation:
-                    self.add_annotation(argument.annotation)
-                    delattr(argument, 'annotation')
-
-        if (
-            hasattr(node.args, 'kwarg')
-            and node.args.kwarg
-            and hasattr(node.args.kwarg, 'annotation')
-            and node.args.kwarg.annotation
-        ):
-            self.add_annotation(node.args.kwarg.annotation)
-            delattr(node.args.kwarg, 'annotation')
-
-        if (
-            hasattr(node.args, 'vararg')
-            and node.args.vararg
-            and hasattr(node.args.vararg, 'annotation')
-            and node.args.vararg.annotation
-        ):
-            self.add_annotation(node.args.vararg.annotation)
-            delattr(node.args.vararg, 'annotation')
+                    argument.annotation = self.add_annotation(argument.annotation)
+
+        for p in ['kwarg', 'vararg']:
+            if (
+                hasattr(node.args, p)
+                and getattr(node.args, p)
+                and hasattr(getattr(node.args, p), 'annotation')
+                and getattr(node.args, p).annotation
+            ):
+                getattr(node.args, p).annotation = self.add_annotation(getattr(node.args, p).annotation)
```

Comment:
This could be simplified to
```suggestion
            if (arg := getattr(node.args, p, None)) and getattr(arg, 'annotation', None):
                arg.annotation = self.add_annotation(arg.annotation)
```
In the end, it's essentially the same thing though.

Reply:
Yeah, I double checked the package classifiers as well before commenting, ~~since I tend to use the walrus operator everywhere and only later notice that I broke compatibility when CI complains~~ 😄